dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/PoolingLayer.java

```diff
@@ -76,19 +81,75 @@ public boolean isLearnable() {
     return false;
   }
 
+  /**
+   * Feedforward function for max pooling.
+   * @param input the input values for this layer.
+   * @return the output values for this layer.
+   */
   private Matrix feedForwardMaxPooling(final Matrix input) {
-    throw new RuntimeException("Not implemented");
+    final int[] inputShape = getInputShape();
+    final Matrix output = matrixFactory.create(outputShape[0] * outputShape[1], input.getColumns());
+    indexMatrix = matrixFactory.create(outputShape[0] * outputShape[1], input.getColumns());
```

Comment:
I think that's not possible since we don't know the number of input images when the class is newly created but we need to `input.getColumns()` to create `indexMatrix`.